### PR TITLE
[SNOW-1368640] Remove qcut fallback when `labels is not False`

### PR DESCRIPTION
**Pull request by sfc-gh-lspiegelberg**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   SNOW-1368640

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Instead of falling back to native pandas, error explicitly out when `labels is not False` for `qcut`.

### CODE REVIEW

File: src/snowflake/snowpark/modin/pandas/general.py
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/general.py:1]: Can you please update the doc string?

**Comment by sfc-gh-helmeleegy** [comment]
Can you also please update `general_supported.rst`?

**Review by sfc-gh-helmeleegy — CHANGES_REQUESTED** [review]
Can you also please update general_supported.rst?

File: src/snowflake/snowpark/modin/pandas/general.py [hunk 1]
```diff
@@ -2192,7 +2190,10 @@ def qcut(
 
     if labels is not False:
         # Labels require categorical, not yet supported. Use native pandas conversion here to compute result.
-        return pandas.qcut(x.to_pandas(), q, **kwargs)
+        ErrorMessage.not_implemented(
+            "Snowpark pandas API supports only labels=False, if you need support"
```
Comment (sfc-gh-nkumar) [inline src/snowflake/snowpark/modin/pandas/general.py:2194]: Can you please update the error message to include method name as well. Something like
```Snowpark pandas qcut method supports only labels=False...```

**Review by sfc-gh-nkumar — COMMENTED** [review]
Looks good. One minor comment to improve error message.

File: src/snowflake/snowpark/modin/pandas/general.py (hunk 1 above)
Comment (sfc-gh-lspiegelberg) [reply src/snowflake/snowpark/modin/pandas/general.py]: absolutely!

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM. Thanks!

**Review by sfc-gh-vbudati — APPROVED** [review]
Looks great, thanks Leonhard!

**Pull request merged**